### PR TITLE
Refactor raw listener

### PR DESCRIPTION
**Pull request by lovebear**
Move entire raw listener function to separate file
Change the switch statements to a object
Remove the raw listener/event all together since it's not needed and
isn't exactly 'raw' anyway

Easier for testing, and maintaining

### CODE REVIEW

File: lib/irc.js [hunk 1]
```diff
@@ -633,6 +140,16 @@ Client.prototype.chanData = function(name, create) {
     return this.chans[key];
 };
 
+Client.prototype._parseCmd = function(message) {
+    var self = this;
+    var fn = parseCmd[message.command];
+    if (fn != undefined) {
+        if (typeof(fn) == 'function') fn(self, message);
```
Comment (jirwin) [inline lib/irc.js]: Instead of passing `self`, we should call the functions in the context of `self`.

File: lib/parse_cmds.js [hunk 2]
```diff
@@ -0,0 +1,543 @@
+// need to set all these locally in function scope
```
Comment (jirwin) [inline lib/parse_cmds.js]: Yup, definitely need to scope these properly to each function.

File: lib/irc.js (hunk 1 above)
Comment (lovebear) [inline lib/irc.js]: what's the advantage?
you would have to replace all the selfs in the original code
also, its adds overhead in performance
which is all ok, but what's the advantage?

File: lib/parse_cmds.js (hunk 2 above)
Comment (lovebear) [inline lib/parse_cmds.js]: It still runs just fine, these can be changed later when tests are added for each function, I just noted it for someone to do later.

**Comment by lovebear** [comment]
Just a note, tests are failing because tests are failing before I made these changes.

**Comment by apeiron** [comment]
I just ran the tests on master and they're passing. Fix your tests, please.

**Comment by Trinitas** [comment]
What about all the self.emit events?

**Comment by lovebear** [comment]
ok my bad, i didn't have this change when i started working on it

https://github.com/martynsmith/node-irc/commit/7a04c41bffe9b89c73622d6a614be9e4936b081b?diff=split

i'll fix it tomorrow

File: lib/parse_cmds.js (hunk 2 above)
Comment (jirwin) [inline lib/parse_cmds.js]: No, they need to be fixed in you PR. Currently in master they are scoped to each event emitted. In your PR, they will be shared across messages that will almost certainly lead to some nasty race conditions.

Comment (lovebear) [inline lib/parse_cmds.js]: There won't be any race condition, it's single threaded, and all these operations are blocking or fire and forget.

Anyway, closing my pull request, Won't have time to do this.
(Most of these changes are just imported from my own project.)

**Pull request closed without merging**